pinot-server/src/main/java/org/apache/pinot/server/api/resources/SegmentMetadataFetcher.java

```diff
@@ -63,9 +77,67 @@ public static String getSegmentMetadata(SegmentDataManager segmentDataManager, L
     }
     ObjectNode segmentMetadataJson = (ObjectNode) segmentMetadata.toJson(columnSet);
     segmentMetadataJson.set("indexes", JsonUtils.objectToJsonNode(getIndexesForSegmentColumns(segmentDataManager)));
+    segmentMetadataJson.set(STAR_TREE_INDEX_KEY, JsonUtils.objectToJsonNode((getStartreeIndexForSegmentColumns(segmentDataManager))));
     return JsonUtils.objectToString(segmentMetadataJson);
   }
 
+  /**
+   * Get the JSON object containing star tree index details for a segment.
+   */
+  private static List<Map<String, Object>> getStartreeIndexForSegmentColumns(SegmentDataManager segmentDataManager) {
+    List<Map<String, Object>> startreeDetails = new ArrayList<>();
+
+    if (segmentDataManager instanceof ImmutableSegmentDataManager) {
```

Comment:
We don't need these `instanceof` checks. The method is declared in the interface

Reply:
This is done.